proposals/NNNN-Vehicle-data-refactoring.md

```diff
@@ -0,0 +1,424 @@
+We need to refactor existing vehicle data items to enhance and/or remove redundancy in vehicle data items. Following are the details on all the changes proposed.
+
+#### Updates to existing vehicle data items:
+
+1. Remove `externalTemperature`.
+   * `externalTemperature` is available in new vehicle data item `ClimateData`.
+2. Remove `PRNDL` vehicle data item.
+   * `PRNDL` is now available in new vehicle data item `GearStatus` as a parameter `gearDriveMode`.
+
+
+### 1. Remove `externalTemperature`.
+
+`externalTemperature` is now available in new vehicle data item `ClimateData` as a parameter. 
+
+
+#### Updates in MOBILE_API:
+
+
+##### Update enum `VehicleDataType`: 
+
+```xml
+<element name="VEHICLEDATA_EXTERNTEMP" since="2.0" until="X.x"/>
+<history>
+	<element name="VEHICLEDATA_EXTERNTEMP"/>
+</history>
+```
+
+
+##### Update function `SubscribeVehicleData` request: 
+
+```xml
+<param name="externalTemperature" type="Boolean" mandatory="false" since="2.0" until="X.x">
+	<description>The external temperature in degrees celsius</description>
+	<history>
+		<param name="externalTemperature" type="Boolean" mandatory="false"/>
+	</history>
+</param>
+```
+
+##### Update function `SubscribeVehicleData` response: 
+
+```xml
+<param name="externalTemperature" type="VehicleDataResult" mandatory="false" since="2.0" until="X.x">
+	<description>The external temperature in degrees celsius</description>
+	<history>
+		<param name="externalTemperature" type="VehicleDataResult" mandatory="false"/>
+	</history>
+</param>
+```
+
+##### Update function `UnsubscribeVehicleData` request: 
+
+```xml
+<param name="externalTemperature" type="Boolean" mandatory="false" since="2.0" until="X.x">
+	<description>The external temperature in degrees celsius</description>
+	<history>
+		<param name="externalTemperature" type="Boolean" mandatory="false"/>
+	</history>
+</param>
+```
+
+##### Update function `UnsubscribeVehicleData` response: 
+
+```xml
+<param name="externalTemperature" type="VehicleDataResult" mandatory="false" since="2.0" until="X.x">
+	<description>The external temperature in degrees celsius</description>
+	<history>
+		<param name="externalTemperature" type="VehicleDataResult" mandatory="false"/>
+	</history>
+</param>
+```
+
+##### Update function `GetVehicleData` request: 
+
+```xml
+<param name="externalTemperature" type="Boolean" mandatory="false" since="2.0" until="X.x">
+	<description>The external temperature in degrees celsius</description>
+	<history>
+		<param name="externalTemperature" type="Boolean" mandatory="false"/>
+	</history>
+</param>
+```
+
+##### Update function `GetVehicleData` response: 
+
+```xml
+<param name="externalTemperature" type="Float" minvalue="-40" maxvalue="100" mandatory="false" since="2.0" until="X.x">
+	<description>The external temperature in degrees celsius</description>
+	<history>
+		<param name="externalTemperature" type="Float" minvalue="-40" maxvalue="100" mandatory="false"/>
+	</history>
+</param>
+```
+
```

Comment:
Can this section be re-organized to be a bit cleaner and easier to read?  Perhaps group all requests together and all responses together?

For example:

Add the following parameter to these function requests:
- `SubscribeVehicleData`
- `UnsubscribeVehicleData`
...

```xml
<param name="externalTemperature" type="Boolean" mandatory="false" since="2.0" until="X.x">
	<description>The external temperature in degrees celsius</description>
	<history>
		<param name="externalTemperature" type="Boolean" mandatory="false"/>
	</history>
</param>
```
Add the following parameter to these function responses:
- `SubscribeVehicleData`
- `UnsubscribeVehicleData`
...

```xml
<param name="externalTemperature" type="VehicleDataResult" mandatory="false" since="2.0" until="X.x">
	<description>The external temperature in degrees celsius</description>
	<history>
		<param name="externalTemperature" type="VehicleDataResult" mandatory="false"/>
	</history>
</param>
```